### PR TITLE
Align data collators across DPO / SFT / Reward / KTO

### PR DESCRIPTION
**Pull request by qgallouedec**
Consistency pass over the data collators so the same thing is written the same way everywhere. No behavior change.

- **Docstrings**: unified intro lines, arg wording (`max_length`, `truncation_mode`, `return_tensors`) and the `# special case for Qwen2.5-VL` comment. Added the missing `Examples` blocks to the KTO collators.
- **Naming**: KTO `batch` → `output`, `ex` → `example`, `torch.int64` → `torch.long`.
- **Structure**: KTO vision now uses inline `"token_type_ids" in processed_prompts` checks (dropping the `has_tti`/`has_mm_tti` locals) so its flush-left / truncate / output blocks match DPO/SFT word-for-word; simplified `mm_token_type_ids` handling to match.
- Fixed a misplaced comma in the repeated BOS comment.

Deeper output-key naming/semantics (e.g. KTO's `completion_input_ids` holds prompt+completion) is left for a follow-up PR.

---

> [!NOTE]
> **Low Risk**
> Mostly docstrings, naming, and structural parity; the KTO vision mm_token_type_ids simplification aligns with DPO/SFT but could differ if a processor ever returned completion-side mm_token_type_ids.
> 
> **Overview**
> This PR standardizes **DPO**, **SFT**, and **KTO** data collators so shared patterns read and are structured the same way; the author describes it as documentation and style alignment with **no intended behavior change**.
> 
> **Documentation** is brought in line across trainers: unified `max_length`, `truncation_mode`, and `return_tensors` wording; richer KTO collator docstrings (including runnable **Examples** for text and vision); a `# Truncate if necessary` comment on DPO vision; and a small fix to the repeated-BOS comment (`BOS twice`).
> 
> **KTO text collator** renames `batch` → `output` and `ex` → `example`, documents optional `ref_logps` / `label` outputs, and adds an explicit per-sequence truncate comment before padding.
> 
> **KTO vision collator** drops `has_tti` / `has_mm_tti` locals in favor of inline `"token_type_ids" in processed_prompts` checks so flush-left, truncate, and output blocks mirror DPO/SFT. **`mm_token_type_ids`** on the completion side now always uses `torch.zeros_like(completion_ids)` (and the same for KL) instead of reading optional fields from the completion processor output—matching DPO/SFT. KL paths use direct `processed_kl["token_type_ids"]` when present instead of `.get` fallbacks.
> 
> **DPO** gets minor doc reordering (`completion_mask` in the outputs list), shared `max_length` phrasing, and small comment/variable naming tweaks (`chosen_token_type_ids`). **SFT** vision collator gets the same doc/comment alignment and the Qwen2.5-VL comment on `mm_token_type_ids`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit fb7514b8936dc0c5c61c8bcdf369008179575da2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/kto/kto_trainer.py [hunk 1]
```diff
@@ -87,27 +87,69 @@ def get_dataset_column_names(dataset: Dataset | IterableDataset) -> list[str]:
...
+    This collator expects each example in the input list to be a dictionary containing the `"prompt_ids"` and
+    `"completion_ids"` keys (and optionally a `"KL_completion_ids"` key). The collator returns a dictionary containing
+    the following keys for each prefix (`"completion"` and, if present, `"KL_completion"`):
+    - `"{prefix}_input_ids"`: Tensor of input IDs, padded to the maximum length of the batch.
+    - `"{prefix}_attention_mask"`: Tensor of attention mask, padded to the maximum length of the batch.
+    - `"{prefix}_mask"`: Tensor indicating the positions of the completion tokens, padded to the maximum length of the
+        batch.
 
-    When `"KL_completion_ids"` is present, the same three tensors are returned for the (mismatched) KL sequence under
-    the `"KL_input_ids"`, `"KL_attention_mask"` and `"KL_completion_mask"` keys.
```
Comment (albertvillanova) [inline trl/experimental/kto/kto_trainer.py:100]: Isn't this reverting what you previously did in:
- #6182 ?

Comment (qgallouedec) [reply trl/experimental/kto/kto_trainer.py:100]: Good catch. The docstrings and the Examples blocks I added mistakenly use the `old completion_*` names, I'll fix the docstrings/examples in both collators.
